async/src/main/java/org/odk/collect/async/WorkerAdapter.kt

```diff
@@ -7,7 +7,12 @@ import androidx.work.WorkerParameters
 abstract class WorkerAdapter(private val spec: TaskSpec, context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {
 
     override fun doWork(): Result {
-        spec.getTask(applicationContext).run()
-        return Result.success()
+        val completed = spec.getTask(applicationContext).get()
+
+        return if (completed) {
+            Result.success()
```

Comment:
We don't care about `Result.failure()` as the only thing it does is give let us inspect failure counts in following work which we're not doing.